### PR TITLE
Practice branch to help solve testData() problem.

### PR DESCRIPTION
**Pull request by yourboyoch**
So I was trying to integrate the testData function into the testData page and for some reason, nothing comes up when I try to create a conversation from the Practice.txt file. Also, although I had all the code for the random conversation generator commented out they still appear when the site is launched, I'm thinking there's something wrong with the Datastore that stores test Data. I was thinking that might have something to do with our `writeThrough()` function. 
Another possible reason I think `Practice.txt` might be failing to make a conversation is that the file cannot be found. I previously had a `try()` `catch()` block within my `testData()` method that captured all the code with `testData()` in the `try()` block but the `FileNotFoundException` kept getting triggered so maybe my file isn't getting found? The Practice file is within my CodeU-Spring-2018 directory. If anyone needs any clarification just ask I know everything I wrote is kinda jargon right now.

### CODE REVIEW

File: .metadata/.log [hunk 1]
```diff
@@ -0,0 +1,79 @@
+!SESSION 2018-03-26 19:38:04.461 -----------------------------------------------
```
Comment (kevmo314) [inline .metadata/.log]: Can you `.gitignore` the `.metadata` directory?

File: src/main/java/codeu/model/store/basic/DefaultDataStore.java [hunk 2]
```diff
@@ -32,68 +36,111 @@
...
+  }
+  
   public boolean isValid() {
     return true;
   }
 
-  public List<User> getAllUsers() {
+  public List<User> getAllUsers(String fileName) {
+	try{
+		testData(fileName);
```
Comment (kevmo314) [inline src/main/java/codeu/model/store/basic/DefaultDataStore.java]: Won't this read the data every single time `getAllUsers()` is called? We should only read it once ideally.

**Review by kevmo314 — CHANGES_REQUESTED** [review]

**Comment by philipcori** [comment]
DefaultDataStore generates the random data and writes it through to PersistentDataStore. The loadUsers() method in PersistantDataStore is what loads the Users, Messages, and Conversations when the app is restarted. To clear all stored data though, just run mvn clean.

File: .metadata/.log (hunk 1 above)
Comment (yourboyoch) [reply .metadata/.log]: don't worry Kevin I'm never merging this branch, I just wanted to get the code up so you could look at it

Comment (yourboyoch) [reply .metadata/.log]: :(( actually I forgot to stash my changes before I switched branches so I might have to copy and paste this after all.

**Pull request closed without merging**